fairseq/tasks/hubert_pretraining.py

```diff
@@ -107,19 +109,21 @@ class HubertPretrainingTask(FairseqTask):
     def __init__(
         self,
         cfg: HubertPretrainingConfig,
-        dictionaries: Dict[str, Dictionary],
     ) -> None:
         super().__init__(cfg)
 
         logger.info(f"current directory is {os.getcwd()}")
         logger.info(f"HubertPretrainingTask Config {cfg}")
-        self._dictionaries = dictionaries
+
+        self.fine_tuning = cfg.fine_tuning
+
+        if cfg.fine_tuning:
+            self.state.add_factory("target_dictionary", lambda: self.load_dictionaries(cfg)[0])
+        else:
+            self.state.add_factory("dictionaries", lambda: self.load_dictionaries(cfg))
```

Comment:
What will happen here is that sef.load_dictionaries(cfg) will be called right away and the function you're passing to add_factory() is an anonymous function that returns the stored object. May be you can save cfg as member variable (self.cfg = cfg) before calling the function. So then you can pass just the function object.

Alternatively, you can use functools.partial to create a function object with the argument that you can pass to add_factory().